File: app/src/main/java/com/example/lucas/haushaltsmanager/Entities/Time.java
package com.example.lucas.haushaltsmanager.Entities;

import java.util.Calendar;
import java.util.Locale;

public class Time {
    private int hour;
    private int minute;

    public Time(int hour, int minute) {

        this.hour = hour;
        this.minute = minute;
    }

    public static Time fromString(String stringTime) {
        int hour = Integer.parseInt(stringTime.substring(0, 2));
        int minute = Integer.parseInt(stringTime.substring(3, 5));

        return new Time(hour, minute);
    }

    @Override
    public String toString() {
        return String.format(Locale.US, "%02d:%02d", hour, minute);
    }

    public long toMillis() {
        Calendar time = Calendar.getInstance();
        time.set(Calendar.HOUR_OF_DAY, hour);
        time.set(Calendar.MINUTE, minute);
        time.set(Calendar.SECOND, 0);

        return time.getTimeInMillis();
    }
}

File: app/src/main/java/com/example/lucas/haushaltsmanager/entities/Time.kt
package com.example.lucas.haushaltsmanager.entities

import java.util.*

class Time(val hour: Int, val minute: Int) {
    companion object {
        @JvmStatic
        fun fromString(stringTime: String): Time {
            val hour = stringTime.substring(0, 2).toInt()
            val minute = stringTime.substring(3, 5).toInt()

            return Time(hour, minute)
        }
    }

    override fun toString(): String {
        return String.format(Locale.US, "%02d:%02d", hour, minute)
    }

    fun toMillis(): Long {
        val time = Calendar.getInstance()
        time[Calendar.HOUR_OF_DAY] = hour
        time[Calendar.MINUTE] = minute
        time[Calendar.SECOND] = 0

        return time.timeInMillis
    }
}